### ISSUE TITLE
Issue with AutoMapper

### ISSUE THREAD

**Issue by iEAmi** [fluentmigrator/fluentmigrator#1107]
When AutoMapper reference exists in project then dotnet-fm can't load AutoMapper.dll.
This issue raise from the conflict in the version of AutoMapper in project and dotnet-fm, when I use the same AutoMapper version as dotnet-fm the problem gone.
The AutoMapper version in my project was 9.0.0 and in dotnet-fm is 8.0.0, this makes this issue.
This issue makes me crazy.

**Comment by fubar-coder** [comment]
Related to #1016. We're looking for a better solution, but for now, you should really keep migrations in their own assembly with - if possible - no external dependencies at all.

**Comment by jzabroski** [comment]
@iEAmi,

@fubar-coder  is correct.  I've been investigating workarounds, and have spent a preposterous amount of time trying to understand all the different testing pathways with .NET "plugin loading".

This appears to be a general problem with .NET Core CLI Tools infrastructure.  The simplest repro I found is to create a .NET Core CLI Tool called TaskRunner that accepts two parameters: `--task` and `--deps=deps.json` (or `--assembly=, where `--deps=deps.json` is a deps.json file generated by the .NET SDK compiler infrastructure as part of resolving/mapping PackageReference elements to link-time dependencies.  I've been thinking of writing up a mosaic of this issue for the general .NET Community to understand these problems, as I don't think they're well understood.

**Comment by jzabroski** [comment]
@iEAmi My preferred workaround is: 

1. If you're not taking advantage of the dependency injection system in FluentMigrator, just use FluentMigrator.MSBuild and the FluentMigrator legacy command line.  This will handle transitive dependencies correctly.
2. Otherwise, use an [In-process migration runner](https://fluentmigrator.github.io/articles/migration-runners.html), as @fubar-coder  suggests

The fundamental problem is that when you install a .NET Global Tool, it expects the same assembly version everywhere.  This sneaks up on you in several ways:
1. You decide to push common FM Extension Methods into a YourCompany.FluentMigrator.Extensions csproj and publish it to your internal nuget server.
    - 1. Those extensions were built with FluentMigrator 3.2.0.
    - 2. But one of your teams wants to upgrade its YourCompany.YourDatabase.Migrations csproj to FluentMigrator 3.2.1 and still use YourCompany.FluentMigrator.Extensions in your nuget server
    - 3. They can't, without upgrading YourCompany.FluentMigrator.Extensions to 3.2.1, which, in turn, means your Visual Studio Package Manager will complain in all other projects that YourCompany.FluentMigrator.Extensions is _not up to date_, even though _upgrading it may break your other projects_.

The tax on this gets larger as your development teams and number of parallel projects grows.

**Comment by iEAmi** [comment]
Thank you @jzabroski for solution.
I would try to create FM extension as internal nuget as you say.

**Comment by jzabroski** [comment]
@iEAmi Sorry, I meant the opposite - don't create FM extension as internal nuget, or the problems I outlined will show up.  I'm trying to teach you the underlying problem, so that you understand how this problem can happen in any .NET Global Tool CLI project like `dotnet-fm`.

tl;dr: Use the In-Process Migration Runner.
